src/hotspot/share/oops/constantPool.inline.hpp

```diff
@@ -51,6 +52,22 @@ inline int ConstantPool::resolved_field_entries_length() const {
     return cache()->resolved_field_entries_length();
 }
 
+inline ResolvedMethodEntry* ConstantPool::resolved_method_entry_at(int method_index) {
+    return cache()->resolved_method_entry_at(method_index);
+}
+
+inline int ConstantPool::resolved_method_entries_length() const {
+    return cache()->resolved_method_entries_length();
+}
+
+inline oop ConstantPool::appendix_if_resolved(int method_index) const {
+  ResolvedMethodEntry* entry = cache()->resolved_method_entry_at(method_index);
+  if (!entry->has_appendix())
```

Comment:
As per the method on `ConstantPoolCache` if you move the body of this into `ResolvedMethodEntry::appendix_if_resolved` you can call that direct from here.